set.mm

```diff
@@ -373209,6 +373328,139 @@ minus one divided by eight (` ( 2 /L N ) ` = -1^(((N^2)-1)/8) ).
+      CUWFYQYOWLWMUUCUYDUXOUUCUYDHYQYOUUBUWJUUAUYDUWCVLUUAUYEUUBUYDUWFVHUUCUYDU
+      RWNVQWOWPUUCUXQUWPUUCYRYPUUBYRWQEZUUAUUBYRUWDWRPUUAYPWQEZUUBUUAYPUWGWRSWS
+      SUUCUYAUWPUUCUXTDNUUCUVOHZUXJUXSUXHUYHUXJUUSYPGZUXSUYHUXJUXIUXPGZUYIUYHYS
+      UXPUXIUYHYPYRUUCUYGUVOUUCYPUWHWRSUUCUYFUVOUUCYRUWEWRSWSWTUYHUVRUVTUVSUYJU
+      YIQUVOUVRUUCUWBPUUAUVTUUBUVOUWGVHUUBUVSUUAUVOUWDVLUUSYPYRVIVJXAUYHUYIUXSU
+      YHUYIHZUUQYOUYKUUQYOUVOUWKUUCUYIUWAVLUUCUYEUVOUYIUUAUYEUUBUWFSVHUVOUWLUUC
+      UYIUWMVLUUCWGYOIJZUVOUYIUUAUYLUUBUUAYOYOVMVNSVHUYHUYIURVPVOVQVRVSVTSWAWBU
+      XEUXKCDNUVCUWRUXHUXDUXJYEUUQYQIUSUVCUWSUXIYSUVCYHUUSYRMUWNUTUNRWCWDVQSWEU
+      UEUXBUXFQZUWPYTUYMUUCYTUXAUXECNYTUWTUXDUWRAYSUWSWFWHUPPPWIWBXBVQPXCXDYCYL
+      YKCNTZOZBNUQUYNBNUIZYNYCUYOBNUYOYCYDNEHYKCNXEXFVTYAUYPYBYAYKCXGTZBXGUIZUY
+      QBNUIZUYPABCXHNXGXKZUYRUYSOXIUYQBNXGBNXLBXGXLXJXMUYNUYQBNUYTUYNUYQOXIYKCN
+      XGXNXMXOXPSYLUYNBNXQXRYLBNXSXT $.
+  $}
+
+  ${
+    $d P a b $.
+    $( There exists a unique decomposition of a prime of the form ` 4 k + 1 `
+       as a sum of squares of two different positive integers.  (Contributed by
+       AV, 4-Jun-2023.)  Spezialisation to different integers, proposed by GL.
+       (Revised by AV, 11-Jun-2023.) $)
+    2sqreunnlt $p |- ( ( P e. Prime /\ ( P mod 4 ) = 1 )
+                       -> E! a e. NN E! b e. NN ( a < b
+                                        /\ ( ( a ^ 2 ) + ( b ^ 2 ) ) = P ) ) $=
+      ( cprime wcel c4 cmo co c1 wceq wa wbr c2 cexp caddc cn wreu wb wi cmul
+      cv cle clt 2sqreunn wne oveq1 oveq2d adantr cc sqcld 2times eqcomd adantl
+      nncn syl ad2antrl eqtrd eqeq1d eleq1 anbi12d cn0 nnz 2nn0 sylancl 2mulprm
+      cz zexpcl oveq2 2t1e2 syl6eq oveq1d crp cc0 2re 4nn nnrp ax-mp 0le2 modid
+      cr 2lt4 mp4an 1ne2 eqcom eqneqall com12 syl5bi syl6bi impcomd expd eqcoms
+      com34 com14 imp31 sylbid expimpd 2a1 pm2.61ine pm4.71d nnre syl2an bibi2d
+      ltlen mpbird ex pm5.32rd reubidva mpbid ) ADEZAFGHZIJZKZBUAZCUAZUBLZXMMNH
+      ZXNMNHZOHZAJZKZCPQZBPQXMXNUCLZXSKZCPQZBPQABCUDXLYAYDBPXLXMPEZKZXTYCCPYFXN
+      PEZKZXSXOYBYHXSXOYBRZYHXSKZYIXOXOXNXMUEZKZRZYJXOYKYJXOYKSZSXNXMXNXMJZYHXS
+      YNYOYHKZXSMXPTHZAJZYNYPXRYQAYPXRXPXPOHZYQYOXRYSJYHYOXQXPXPOXNXMMNUFUGUHYF
+      YSYQJZYOYGYEYTXLYEXPUIEZYTYEXMXMUNUJUUAYQYSXPUKULUOUMUPUQURYFYRYNSZYOYGXI
+      XKYEUUBYRXKYEXIYNXKYEXIYNSSSAYQAYQJZXKXIYEYNUUCXKXIYEYNSZUUCXKXIKYQFGHZIJ
+      ZYQDEZKZUUDUUCXKUUFXIUUGUUCXJUUEIAYQFGUFURAYQDUSUTYEUUHYNYEUUGUUFYNYEUUGX
+      PIJZUUFYNSYEXPVFEZUUGUUIRYEXMVFEMVAEUUJXMVBVCXMMVGVDXPVEUOUUIUUFMIJZYNUUI
+      UUEMIUUIUUEMFGHZMUUIYQMFGUUIYQMITHMXPIMTVHVIVJVKMVTEFVLEZVMMUBLMFUCLUULMJ
+      VNFPEUUMVOFVPVQVRWAMFVSWBVJURIMUEZUUKYNSWCUUKIMJZUUNYNMIWDUUOUUNYNYNIMWEW
+      FWGVQWHWHWIWFWHWJWLWKWMWNUPWOWPYKYJXOWQWRWSYHYIYMRXSYHYBYLXOYFXMVTEZXNVTE
+      YBYLRYGYEUUPXLXMWTUMXNWTXMXNXCXAXBUHXDXEXFXGXGXH $.
+
+    $( There exists a unique decomposition of an odd prime as a sum of squares
```

Comment:
Why is 2 explicitly ruled out here?  It seems that the condition 4k + 1 is sufficient.